src/StdChains.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2 <0.9.0;
+
+pragma experimental ABIEncoderV2;
+
+import "./Vm.sol";
+
+abstract contract StdChains {
+    VmSafe private constant vm = VmSafe(address(uint160(uint256(keccak256("hevm cheat code")))));
+
+    struct Chain {
+        // The chain name, using underscores as the separator to match `foundry.toml` conventions.
+        string name;
+        // The chain's Chain ID.
+        uint256 chainId;
+        // A default RPC endpoint for this chain.
+        // NOTE: This default RPC URL is included for convenience to facilitate quick tests and
+        // experimentation. Do not use this RPC URL for production test suites, CI, or other heavy
+        // usage as you will be throttled and this is a disservice to others who need this endpoint.
+        string rpcUrl;
+    }
+
+    // Maps from a chain's name (matching what's in the `foundry.toml` file) to chain data.
+    mapping(string => Chain) private chains;
+
+    function getChain(string memory name) internal virtual returns (Chain memory) {
+        initialize();
+        return chains[name];
+    }
+
+    function setChain(string memory name, uint256 chainId, string memory rpcUrl) internal virtual {
+        chains[name] = Chain(name, chainId, rpcUrl);
+    }
+
+    bool private initialized;
+
+    function initialize() private {
+        if (initialized) return;
+
+        chains["anvil"] = Chain("Anvil", 31337, "http://127.0.0.1:8545");
```

Comment:
Probably should use the `setChain` method here, especially if that might write to two mappings now instead of one